### PR TITLE
Add age filter and delete action for CloudWatch Alarms

### PR DESCRIPTION
**Pull request by calvinhp**
I have one last test to finish for the age filter, I need to wait a day so it can correctly pick up the filter I created today.

### CODE REVIEW

**Comment by kapilt** [comment]
Age filters are built in to the value filter...

**Comment by kapilt** [comment]
also please note that you have 111 mostly data files in this pull request, ideally look trim the number of extant alarms pre record to reduce the extraneous data.

**Comment by calvinhp** [comment]
> also please note that you have 111 mostly data files in this pull request, ideally trim the number of extant alarms pre record to reduce the extraneous data.

I want to test that we can delete 101 alarms in 2 API calls.  Should I only include the run of the policy in the test and not the setup?

**Comment by calvinhp** [comment]
> Age filters are built in to the value filter, i think you can just skip that and use the value filter with value_type: age, age can be done as fractional days re wait a day.

The `AgeFilter` says I need to override the `date_attribute` in a subclass.  Did I miss something, or are you pointing at another issue with this implementation?

**Comment by kapilt** [comment]
separate date filters are legacy, we have generic date filters builtin to the value type filter which can do date comparisons on any attribute.

**Comment by kapilt** [comment]
the test should only delete 1 ideally.

File: c7n/resources/cw.py [hunk 1]
```diff
@@ -35,6 +37,71 @@ class resource_type(object):
...
+    """
+
+    schema = type_schema('delete')
+    permissions = ('cloudwatch:DeleteAlarms',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('cloudwatch')
+        alarms = []
+
+        for i, r in enumerate(resources, start=1):
```
Comment (kapilt) [inline c7n/resources/cw.py]: for resource_set in chunks(resources, 100): client.delete_alarms(AlarmNames=[r['Name'] for r in resource_set])

File: c7n/filters/core.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 import operator
 import re
 
+import pytz
```
Comment (kapilt) [inline c7n/filters/core.py]: this is adding in a redundant dependency.

File: c7n/filters/core.py [hunk 3]
```diff
@@ -349,6 +350,11 @@ def process_value_type(self, sentinel, value):
                     value = parse(value)
                 except (AttributeError, TypeError):
                     value = 0
+
+            # Work around placebo issues with tz
+            if isinstance(value, datetime) and not value.tzinfo:
+                value = pytz.utc.localize(value)
```
Comment (kapilt) [inline c7n/filters/core.py]: i'd say just use value.replace(tzinfo=tzutc()) although i'm missing exact context for the comparison (tzutc().fromutc(dt) will convert to local time), might be worth linking the upstream pull request https://github.com/garnaat/placebo/pull/50

**Comment by kapilt** [comment]
there's still way too much test data for whats needed here, like a 100x more on the describes, ideally this could just create one alarm in test setup, and delete one alarm.

there's some urgency on this so i'm going to go ahead and clean this up for merging.

**Pull request closed without merging**